src/components/Carousel.test.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import Carousel from './Carousel';
+
+describe('<Carousel/>', () => {
+  let wrapper, props;
+
+  beforeEach(() => {
+    props = {
+      images: [
+        {
+          id: '123yvu1231bv23',
+          url: 'https://placeholder.com/480x200.png',
+          description: 'brief image description here'
+        },
+        {
+          id: '09nhgfnbdv212j',
+          url: 'https://placeholder.com/480x250.png',
+          description: 'brief image description here'
+        }
+      ]
+    };
+    wrapper = shallow(<Carousel {...props} />);
+  });
+  it('renders all images and only once', () => {
+    let renderedImages;
+    props.images.forEach(image => {
+      renderedImages = wrapper
+        .findWhere(child => {
```

Comment:
i'm not crazy about this... rather tough to read. i think it's the chainign of all these methods... `.forEach().findWhere().map()` i'll try to find you an example of another way to do it or Maybe we can pair on another solution.